worlds/sm/__init__.py

```diff
@@ -122,9 +122,8 @@ def __init__(self, world: MultiWorld, player: int):
 
     @classmethod
     def stage_assert_generate(cls, multiworld: MultiWorld):
-        rom_file = get_base_rom_path()
-        if not os.path.exists(rom_file):
-            raise FileNotFoundError(rom_file)
+        # don't need rom anymore
+        pass
 
```

Comment:
Since this method isn't used anymore, it can probably just be removed due to it not being required
https://github.com/ArchipelagoMW/Archipelago/blob/0265f4d80947d02bf2b8b261f3c3ca469c117056/docs/world%20api.md?plain=1#L440-L444